imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,346 @@
+  productVariantId: ID!
+
+  "The product's slug"
+  productSlug: String
+
+  "A title for use in cart/orders that conveys the selected product's title + chosen options"
+  title: String!
+}
+
+enum CartReconciliationMode {
+  merge
+  keepAccountCart
+  keepAnonymousCart
+}
+
+####################
+# Queries
+####################
+
+extend type Query {
+  "Finds a cart by the cart ID and anonymous cart token."
+  anonymousCartByCartId(cartId: ID!, token: String!): Cart
+
+  # Access control should ensure that only authenticated users can find their own
+  # cart. Additionally, administrative roles can find carts.
+  # Shop ID is necessary if shops have separate carts but share the same account pool
+  "Find a cart for a given account ID."
+  accountCartByAccountId(accountId: ID!, shopId: ID!): Cart
+}
+
+####################
+# Inputs
+####################
+
+input CreateCartInput {
+  "Array of items to add to new cart."
+  items: [CartItemInput]!  # NOTE: Don't allow creation of an empty cart
+
+  "ShopId association for the cart."
+  shopId: ID! # NOTE: We'll require shopId though it's not in current code.
+
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
+}
+
+# Input for adding item(s) to a cart.
+# Do we need catalog item information? Maybe do this as part of the "analytics" event tracking.
+input CartItemInput {
+  "Arbitrary additional metadata about this cart item."
+  metafields: [MetafieldInput]
+
+  "The product and chosen options"
+  productConfiguration: ProductConfigurationInput!
+
+  quantity: Int!
+
+  "ShopId from which the product is added."
+  shopId: ID!
+
+  # Server to ensure that the client price is the same as the server price
+  # Inform customer that prices are different and offer to "add anyway" or "cancel" if not.
+  "The current price of this item"
+  price: Float!
+
+  "The currency for the `price` and `priceWhenAdded` fields"
+  priceCurrency: String!
+}
+
+input ProductConfigurationInput {
+  "The Product ID"
+  productId: ID!
+
+  "The ProductVariant ID"
+  productVariantId: ID!
+}
+
+input AddCartItemsInput {
+  "The cart ID"
+  cartId: ID!
+
+  "Array of items to be added to the cart"
+  items: [CartItemInput]!
+
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
+}
+
+input RemoveCartItemsInput {
+  "The cart ID"
+  cartId: ID!,
+
+  "Array of items to remove from the cart."
+  cartItemIds: [ID]!
+
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
+}
+
+input UpdateCartItemInput {
+  "The cart item ID"
+  cartItemId: ID!
+
+  "New absolute value for specified cart item's quantity. Not an incremental value."
+  quantity: Int!
+
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
```

Comment:
I guess `clientMutationId` would not be needed in this input because it's only used within another input that already has it.